### PR TITLE
Wrap DB queries inside manager to enable rollback

### PR DESCRIPTION
**Pull request by gsmithun4**
Resolves https://github.com/ToolJet/ToolJet/issues/3808

### CODE REVIEW

File: server/test/services/group_permissions.service.spec.ts [hunk 1]
```diff
@@ -27,11 +27,7 @@ describe('GroupPermissionsService', () => {
     it('should validate uniqueness of group permission group name', async () => {
       const { adminUser } = await setupOrganization(nestApp);
 
-      const data = await service.create(adminUser, 'avengers');
-
-      expect(data.id).toBeDefined();
```
Comment (akshaysasidrn) [inline server/test/services/group_permissions.service.spec.ts:32]: Does this fail now?

File: server/src/helpers/utils.helper.ts [hunk 2]
```diff
@@ -52,3 +53,13 @@ export function sanitizeInput(value: string) {
 export function lowercaseString(value: string) {
   return value?.toLowerCase();
 }
+
+export async function dbManagerWrapper(operation: (...args) => any, manager: EntityManager): Promise<any> {
```
Comment (akshaysasidrn) [inline server/src/helpers/utils.helper.ts:57]: Can we reword this from `dbManagerWrapper` to `dbTransactionWrap` ?

Does refactoring this to the one below work?

```
export async function dbManagerWrapper(operation: (...args) => any, manager = getManager()): Promise<any> {
  await manager.transaction(async (manager) => {
    return await operation(manager);
  });
 }
```
The advantage being one wouldn't have to always pass the second param and have it wrapped in a transaction

**Comment by akshaysasidrn** [comment]
Please check the failing e2e cases 🙇🏾

File: server/test/services/group_permissions.service.spec.ts (hunk 1 above)
Comment (gsmithun4) [reply server/test/services/group_permissions.service.spec.ts:32]: for update request we are not expecting any response params in front end

Comment (gsmithun4) [reply server/test/services/group_permissions.service.spec.ts:32]: so returning empty response

File: server/src/helpers/utils.helper.ts (hunk 2 above)
Comment (gsmithun4) [reply server/src/helpers/utils.helper.ts:57]: Not working, if `manager` exists we have already started transactions for previous queries  `await manager.transaction` throws error

Comment (akshaysasidrn) [reply server/src/helpers/utils.helper.ts]: Okay, thought it would result in nested transactions 🤔 
Ignore then 👍🏾

**Review by akshaysasidrn — APPROVED** [review]

**Pull request merged**